lib/developer_portal/app/controllers/developer_portal/login_controller.rb

```diff
@@ -45,6 +49,12 @@ def destroy
 
   private
 
+  def render_bot_error
+    @session = Session.new
+    assign_drops add_authentication_drops
+    render action: :new
```

Comment:
I don't understand these lines :grimacing: 
But I can see this is the same thing as `render_creation_error`. Which makes me think... maybe we could merge those two to avoid duplication?

e.g.

```
  def render_login_error(error_message = nil)
    @session = Session.new
    flash.now[:error] = error_message if error_message
    assign_drops add_authentication_drops
    render action: :new
  end
```

And then call it like this (according to the place):
```
  render_login_error

  render_login_error(@strategy.error_message)
```

Reply:
good point

Reply:
Done in https://github.com/3scale/porta/pull/3659/commits/7ae3497c9aa70b28061db989d5381309744dd49c